From 17902edf938e2dc181167a75fd91487e10dfd9a5 Mon Sep 17 00:00:00 2001
Message-ID: <17902edf938e2dc181167a75fd91487e10dfd9a5.1729453324.git.sharlatanus@gmail.com>
In-Reply-To: <cover.1729453324.git.sharlatanus@gmail.com>
References: <cover.1729453324.git.sharlatanus@gmail.com>
From: Sharlatan Hellseher <sharlatanus@gmail.com>
Date: Thu, 10 Oct 2024 11:06:31 +0100
Subject: [PATCH 16/42] gnu: python-pyregion: Update to 2.3.0.

* gnu/packages/astronomy.scm (python-pyregion): Update to 2.3.0.
[arguments] <#:phases>: Add 'create-setup.py phase.
[native-inputs]: Add python-extension-helpers, python-pytest-astropy, and
python-setuptools.

Change-Id: I4c114a5ec51a7dae2197f60579f1e63597c23ea7
---
 gnu/packages/astronomy.scm | 20 +++++++++++++++++---
 1 file changed, 17 insertions(+), 3 deletions(-)

diff --git a/gnu/packages/astronomy.scm b/gnu/packages/astronomy.scm
index 2d37ae3999..00a7e0d17b 100644
--- a/gnu/packages/astronomy.scm
+++ b/gnu/packages/astronomy.scm
@@ -5233,28 +5233,42 @@ (define-public python-pynbody
 (define-public python-pyregion
   (package
     (name "python-pyregion")
-    (version "2.2.0")
+    (version "2.3.0")
     (source
      (origin
        (method url-fetch)
        (uri (pypi-uri "pyregion" version))
        (sha256
-        (base32 "0l7qb7r8fnv46mdih4m5b8jaxixgpw6m7v37dpikjkblgh0vigaw"))))
+        (base32 "09a98v3zk1vdjns1q64al58mapr4cns3nlnyi6b26wqi888qfjg8"))))
     (build-system pyproject-build-system)
     (arguments
      (list
       #:phases
       #~(modify-phases %standard-phases
+          ;; setup.py was removed in b26ec4fe88e29447dc8391fcdef7082a4f7876ce
+          ;; TODO: Check how to implement it in python-build-system.
+          (add-after 'unpack 'create-setup.py
+            (lambda _
+              (call-with-output-file "setup.py"
+                (lambda (port)
+                  (format port "from setuptools import setup
+from extension_helpers import get_extensions
+setup(ext_modules=get_extensions())")))))
           (add-before 'check 'build-extensions
             (lambda _
               ;; Cython extensions have to be built before running the tests.
               (invoke "python" "setup.py" "build_ext" "--inplace"))))))
     (propagated-inputs
-     (list python-astropy python-numpy python-pyparsing))
+     (list python-astropy
+           python-numpy
+           python-pyparsing))
     (native-inputs
      (list python-cython
+           python-extension-helpers
            python-pytest
+           python-pytest-astropy
            python-pytest-astropy-header
+           python-setuptools
            python-setuptools-scm))
     (home-page "https://github.com/astropy/pyregion")
     (synopsis "Python parser for ds9 region files")
-- 
2.46.0

